### PR TITLE
Expand security section and prune dead link

### PR DESCRIPTION
**Pull request by Hoverbear**
### What is changed?

Adds some clarification/working/tone changes around TLS setup and prunes out a dead link for a simple inline script.

### Any related PRs or issues?

Nope

### Which version does your change affect?

3.0

### CODE REVIEW

File: content/docs/3.0/tasks/configure/security.md [hunk 1]
```diff
@@ -15,17 +15,70 @@ This page discusses how to secure your TiKV deployment. Learn how to:
...
+Transport Layer Security is an standard protocol for protecting networking communications from tampering or inspection. TiKV uses OpenSSL, an industry standard, to implement it's TLS encryption.
 
-To use TLS encryption with TiKV, you need to prepare the certificates for the deployment.
+It's often necessary to use TLS in situations where TiKV is being deployed or accessed from outside of a secure virtual local area network (VLAN).  This includes deployments which cross the WAN (the public internet), are part of an untrusted datacenter network, or where other untrustworthy users or services are active.
 
-### Prepare Certificates
+Before you get started, review your infrastructure. Your organization may already use something like the [Kubernetes certificates API](https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/) to issue certificates. You will need a **Certificate Authority** certificate, a unique **certificates** and **keys** for each TiKV or PD service, and one or many certificates for TiKV clients depending on your needs. If you have these, you can skip the optional section below.
```
Comment (dcalvin) [inline content/docs/3.0/tasks/configure/security.md]: > a **Certificate Authority** certificate, a unique **certificates** and **keys** for each TiKV or PD service

Not sure I follow here - a unique **certificates** and **keys** equal to  a **Certificate Authority** certificate?
```

File: content/docs/3.0/tasks/configure/security.md [hunk 2]
```diff
@@ -15,17 +15,70 @@ This page discusses how to secure your TiKV deployment. Learn how to:
...
+
+If your organization doesn't yet have a public key infrastructure (PKI) you can create a simple Certificate Authority then issue certificates for the services in your deployment. Below, you can see how to do this in a few relatively quick steps.
```
Comment (dcalvin) [inline content/docs/3.0/tasks/configure/security.md]:
```suggestion
If your organization doesn't yet have a public key infrastructure (PKI), you can create a simple Certificate Authority to issue certificates for the services in your deployment. The instructions below show you how to do this in a few quick steps:
```

File: content/docs/3.0/tasks/configure/security.md [hunk 3]
```diff
@@ -15,17 +15,70 @@ This page discusses how to secure your TiKV deployment. Learn how to:
...
+for i in $(seq 1 $NUM_TIKV_NODES); do
+    ./easyrsa gen-req tikv$i nopass
+    ./easyrsa sign-req server tikv$i
+done
+
+./easyrsa gen-req client nopass
+./easyrsa sign-req server client
+```
+
+If you run this script you'll need to interactively answer some questions and make some confirmations, you can answer anything for the CA common name, and for the PD and TiKV nodes you should use the hostnames.
```
Comment (dcalvin) [inline content/docs/3.0/tasks/configure/security.md]:
```suggestion
If you run this script, you'll need to interactively answer some questions and make some confirmations. You can answer with anything for the CA common name.  For the PD and TiKV nodes, you should use the hostnames.
```

File: content/docs/3.0/tasks/configure/security.md [hunk 4]
```diff
@@ -15,17 +15,70 @@ This page discusses how to secure your TiKV deployment. Learn how to:
...
+
+{{< info >}}
+You can explore the `easyrsa/vars.example` file if you are hoping to write an unattended script.
+{{< /info >}}
+
+After, you should have something like this:
```
Comment (dcalvin) [inline content/docs/3.0/tasks/configure/security.md]:
```suggestion
If the script runs successfully, you should have something like this:
```

File: content/docs/3.0/tasks/configure/security.md (hunk 1 above)
Comment (Hoverbear) [reply content/docs/3.0/tasks/configure/security.md]: I think I found a more clear way to say this :)

File: content/docs/3.0/tasks/configure/security.md [hunk 5]
```diff
@@ -15,17 +15,70 @@ This page discusses how to secure your TiKV deployment. Learn how to:
...
+for i in $(seq 1 $NUM_TIKV_NODES); do
+    ./easyrsa gen-req tikv$i nopass
+    ./easyrsa sign-req server tikv$i
+done
+
+./easyrsa gen-req client nopass
+./easyrsa sign-req server client
+```
+
+If you run this script, you'll need to interactively answer some questions and make some confirmations. You can answer with anything for the CA common name.  For the PD and TiKV nodes, you should use the hostnames.
```
Comment (dcalvin) [inline content/docs/3.0/tasks/configure/security.md]:
```suggestion
If you run this script, you'll need to interactively answer some questions and make some confirmations. You can answer with anything for the CA common name. For the PD and TiKV nodes, use the hostnames.
```

File: content/docs/3.0/tasks/configure/security.md [hunk 6]
```diff
@@ -15,17 +15,70 @@ This page discusses how to secure your TiKV deployment. Learn how to:
...
+Transport Layer Security is an standard protocol for protecting networking communications from tampering or inspection. TiKV uses OpenSSL, an industry standard, to implement it's TLS encryption.
 
-To use TLS encryption with TiKV, you need to prepare the certificates for the deployment.
+It's often necessary to use TLS in situations where TiKV is being deployed or accessed from outside of a secure virtual local area network (VLAN). This includes deployments which cross the WAN (the public internet), which are part of an untrusted data center network, or where other untrustworthy users or services are active.
 
-### Prepare Certificates
+Before you get started, review your infrastructure. Your organization may already use something like the [Kubernetes certificates API](https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/) to issue certificates. You will need a **Certificate Authority** (CA) certificate, the key for that CA certificate, as well as individual unique **certificates**, and **keys** for each TiKV or PD service, and one or many certificates for TiKV clients depending on your needs. If you have these, you can skip the optional section below.
```
Comment (dcalvin) [inline content/docs/3.0/tasks/configure/security.md]:
```suggestion
## Before you get started

Before you get started, review your infrastructure. Your organization may already use something like the [Kubernetes certificates API](https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/) to issue certificates. You will need to verify the following in your deployment:

-  **Certificate Authority** (CA) certificate and the key for it
-  Individual unique **certificates** and **keys** for each TiKV or PD service
-  One or many certificates for TiKV clients depending on your needs

 If you have these, you can skip the optional section below.
```

File: content/docs/3.0/tasks/configure/security.md [hunk 7]
```diff
@@ -15,17 +15,70 @@ This page discusses how to secure your TiKV deployment. Learn how to:
...
+
```
Comment (dcalvin) [inline content/docs/3.0/tasks/configure/security.md]:
```suggestion
## Configure TLS for TiKV

```

Comment (Hoverbear) [reply content/docs/3.0/tasks/configure/security.md]: This seems like a duplicate of the above heading?

Comment (dcalvin) [reply content/docs/3.0/tasks/configure/security.md]: Sorry I totally forgot about this PR. This heading is intended as a second level that includes the configuration steps. The one above - actually I suggest we remove it. I looked at other docs in the repo and they don't have a 2nd level heading at the beginning of the document.

File: content/docs/3.0/tasks/configure/security.md [hunk 8]
```diff
@@ -15,17 +15,78 @@ This page discusses how to secure your TiKV deployment. Learn how to:
 
 ## Transport Layer Security (TLS)
```
Comment (dcalvin) [inline content/docs/3.0/tasks/configure/security.md]:
```suggestion
```
Suggest removing this as commented below

File: content/docs/3.0/tasks/configure/security.md (hunk 7 above)
Comment (Hoverbear) [reply content/docs/3.0/tasks/configure/security.md]: The optional section below has nothing to do with configuring TiKV though, it's just how to roll keys for any application.

**Comment by Hoverbear** [comment]
@dcalvin PTAL again?

**Pull request merged**